### ISSUE TITLE
Not Liked 400

### ISSUE THREAD

**Issue by ugurozturk** [ugurozturk/instabot.pyV2#3]
I know this error well. Trying to fix it

**Comment by ugurozturk** [comment]
upgraded requests 2.18.4 > testing

**Comment by ugurozturk** [comment]
[x] updated accept_language > checking

**Comment by ugurozturk** [comment]
It is working me now. Need to know others

**Comment by hgic86** [comment]
I still git this error for some reason with your new code. Please help!

<img width="467" alt="screen shot 2018-01-04 at 9 48 08 am" src="https://user-images.githubusercontent.com/4732519/34568637-6ebfe712-f134-11e7-9cd7-47fb43b69a69.png">

**Comment by ugurozturk** [comment]
@hgic86 how old is the account ? How much do you prefer to like, follow , unfollow, comment per day ?

**Comment by hgic86** [comment]
@ugurozturk my account is about 3 years old. All I do is like 2000 per day. I don't follow, unfollow or comment.

**Comment by ugurozturk** [comment]
@hgic86 probabby this is the reason of error 400. 2000 is so much. The recommended limit is 1000.

**Comment by FreddyFY** [comment]
I'm getting same error "Not liked: 400". Like limit is 750. Comments are working.  On instagram.com everything is working properly.

**Comment by ugurozturk** [comment]
@FreddyFY how old is your account ?

**Comment by trevorgit** [comment]
Please explain why the age of an account has anything to do with this issue? I have tested using the default values you have set in the repo and this is still giving (Not liked: 400)

**Comment by ugurozturk** [comment]
@trevorgit Sure, when the bot is new we do recommending max 100 likes for a day. Then it goes 200 after 1 week, then it can be 300 after 2 week. As you see in the first comment of the issue. it is mine so yeah. I also had trobbling with error 400. Did a lot things via this repo and time passed. My account is 3 or 4 month now. And i can liking / following / unfollowing without any error.

**Comment by bgokden** [comment]
Hi @ugurozturk I got this error once then it continued without any error. 
I use 1000 likes per day without follow/unfollow. Account is very old like 5 years and I am using this bot with the same configuration for more then one year.
Maybe they have added another fraud detection check?

**Comment by ugurozturk** [comment]
@bgokden you can check the errors at which medias too. Just open the database file and there is table named "medias" and in the table, there is column, media_id and status and datetime. So you can check how offen you having the error 400.

Note : We calling error 400 is probabby different than yours. When they are getting the error 400 then they can't like any media, after by after always getting error 400. While the bot sleeps for 2hour then the bot runs few hours normal then they getting error 400 again. Your bot probabby runs stable without stopping nor having troubles on the liking.

**Comment by FreddyFY** [comment]
@ugurozturk my account is about 4 years old. The script was running from August 2017 to Dezember 2017 without any errors. Always the same settings

**Comment by ugurozturk** [comment]
@FreddyFY after that be known i also dont know the solution so lets find the reason together. Let the bot stop for 8hours. Somewhere i did read, instagram could do shadow ban. Then perhaps you can try lowing the number of commenting limit. Mine like limit is 1000 and i dont having any trouble so you can also make it to 1000, but follow / unfollow is 300.

**Comment by mcrowder65** [comment]
Getting this same error.. Had likes on 2400 per day, account is about 1.5 years old.

**Comment by hgic86** [comment]
@ugurozturk I lowered my likes to 1000 and it started working. Thank you! Will you be able to expand the capacity for "likes" eventually?

**Comment by ugurozturk** [comment]
@hgic86 it is not about me or the code. it is about instagram. Perhaps you can try to increase the limit little by little. like setting the limit to 1100 ? So you can report here what happens when you increase it.

**Comment by ugurozturk** [comment]
@mcrowder65 let it take some rest then make it liking 1000 per day. Check the account if everything goes well then you can try to make it increase. But be known i dont have the solution and the real problem whatever causing it to.

**Comment by aruser1** [comment]
Hey, do I have to download this repo and install it normally like we all did in the original instabot.py - pip install ....blabla in cmd ? or just have to extract in a folder? 
Also, can we keep it in a separate folder than the original instabot.py ? PYTHON KNOWLEDGE 2/10

**Comment by ugurozturk** [comment]
@aruser1 please open a new issue for it. Then the others can also get help from it. this issue topid is about not liked 400.

**Comment by barronlroth** [comment]
Also got this issue on DigitalOcean server. Account is 6 years old. 950 likes/day. No comments. Tried again on local machine and got to ~500 likes before error 400 reappeared. Will wait cool-down period and try again in a couple days.

**Comment by ugurozturk** [comment]
@barronlroth so the problem is might be not because of likes/day . What is your follow / unfollow limits ?

**Comment by bgokden** [comment]
By the way, Instagram does not have a likes/day limit, it is more precise. I had limit per 4 hours when they had the old api previously. So you can not do 1000 likes in 8 hours and wait rest of the day.
I realised with the removal of already liked media, 1000 likes/day is already close to the suspicious usage limit and it fails depending on the hashtags you use. In my bots, we have around 50 hashtags, so bot waits between hashtags since it looks at media/check liked. In my calculation my bot requires more than 12 hours to finish 1000 likes at minimum, but some I have checks like minimum and maximum like constraints so some days, it can not even finish liking 1000 likes. So be careful on what you choose for tags, number of tags. These also effects your limit of likes. I don't use follow/unfollow, it is a bit suspicious activity.

**Comment by ugurozturk** [comment]
@bgokden the bot's algorithm is gives the delays like. NextLike_Time = now_Time + 24*60*60 / 1000 +- random_sec . So you are making 1000likes for 24hour. if you open 4hours the bot, then it would 1000/24*4 166likes for the 4 hours. I also made start stop time feature. The bot can start at 7am, and can stop at 23:59pm. If you set 1000likes for a day, with start and stop times. The bot would do 1000/24*(24-7) 708 likes for a day.

**Comment by ugurozturk** [comment]
Going to check the algorithm in real results via database. Will edit this comment

**Comment by bgokden** [comment]
@ugurozturk I see that sometimes, bot waits on other processes too much, so it takes more than what is calculated, and if you have many tags, it goes through querying media, removing already liked media process. I think we did not see this problem before because removing already liked media feature makes liking significantly more faster. Previously bot was liking the same image multiple times. By the way, I think it is a great feature and previously bot was wasting time on the same images if you use a small number of tags or unpopular tags.

**Comment by ugurozturk** [comment]
@bgokden https://github.com/ugurozturk/instabot.pyV2/issues/35 created issue for it

**Comment by barronlroth** [comment]
@ugurozturk not using follow/unfollow or comments, just likes.

**Comment by iceman183** [comment]
@barronlroth Probably blocked ip for too many likes,

**Comment by ugurozturk** [comment]
@barronlroth @iceman183 please use https://github.com/instabot-py/instabot.py i merged my changes to original repo
